pageserver/src/tenant/mgr.rs

```diff
@@ -256,83 +256,105 @@ async fn init_load_generations(
     Ok(Some(generations))
 }
 
+/// Given a directory discovered in the pageserver's tenants/ directory, attempt
+/// to load a tenant config from it.
+///
+/// If file is missing, return Ok(None)
+fn load_tenant_config(
+    conf: &'static PageServerConf,
+    dentry: Utf8DirEntry,
+) -> anyhow::Result<Option<(TenantId, anyhow::Result<LocationConf>)>> {
+    let tenant_dir_path = dentry.path().to_path_buf();
+    if crate::is_temporary(&tenant_dir_path) {
+        info!("Found temporary tenant directory, removing: {tenant_dir_path}");
+        // No need to use safe_remove_tenant_dir_all because this is already
+        // a temporary path
+        if let Err(e) = std::fs::remove_dir_all(&tenant_dir_path) {
+            error!(
```

Comment:
why not return the path to be removed and have the calling logic run the file removal in batch over all paths

Reply:
We're already running with per-tenant parallelism here: passing back up and having another round of parallel `spawn_blocking`s for the deletions wouldn't add much.  It's also an edge case rather than a piece of work we will usually encounter in startup.